### ISSUE TITLE
hazelcast 4.* dependency in micronaut-cache bom

### ISSUE THREAD

**Issue by ckosmowski** [micronaut-projects/micronaut-cache#315]
### Steps to Reproduce

1. Create a micronaut project also using features hazelcast 3.6.* (not for caching, for our own purposes)
2. Update micronaut to 2.4.2

### Expected Behaviour

Everything works as before only with the new micronaut code

### Actual Behaviour

Everything breaks because for some reason the micronaut-cache pulls in a hazelcast-4.* dependecy (without importing micronaut-cache-hazelcast!) which overrides the existing 3.6.* dependecy. Everything is fine at compile time but at runtime there is a "ClassNotFoundException" for hazelcasts NodeContext class because it moved in hazelcast 4.

**Comment by graemerocher** [comment]
this project currently depends on hazelcast 4 where is the existing 3.6 dependency you refer to being pulled from?

**Comment by ckosmowski** [comment]
The existing 3.6. version is a direct dependency of our services. We use hazelcast for locking, distributed maps and so on, not for caching in our case. And we also use hazelcast-zookeeper for discovery and since the zookeeper is a bit out of date, we need to stick with hazelcast 3.* for now.

We worked around this problem by strictly fixing the hazelcast version:

implementation ("com.hazelcast:hazelcast") {
      version {
          strictly("3.12.12")
      }
  }
 (It's now 3.12.12 because we tried a lot before we discovered the hazelcast 4 dependency).
This does not seem to break anything in micronaut. (We ar enot using micronaut-cache-hazelcast)
